### PR TITLE
PIN-3496 Added getTenantByExternalId endpoint

### PR DESCRIPTION
**Pull request by nttdata-rtorsoli**

### CODE REVIEW

File: src/main/scala/it/pagopa/interop/tenantprocess/api/impl/TenantApiServiceImpl.scala [hunk 1]
```diff
@@ -753,4 +753,22 @@ final case class TenantApiServiceImpl(
       updateVerifiedAttributeExtensionDateResponse[Tenant](operationLabel)(updateVerifiedAttributeExtensionDate200)
     }
   }
+
+  override def getTenantByExternalId(origin: String, code: String)(implicit
+    contexts: Seq[(String, String)],
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem],
+    toEntityMarshallerTenant: ToEntityMarshaller[Tenant]
+  ): Route = {
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/tenantprocess/api/impl/TenantApiServiceImpl.scala]:
```suggestion
  ): Route = authorize(ADMIN_ROLE, API_ROLE, SECURITY_ROLE, M2M_ROLE, SUPPORT_ROLE) {
```

File: src/test/resources/authz.json [hunk 2]
```diff
@@ -31,6 +31,17 @@
         "support"
       ]
     },
+    {
+      "route": "getTenantByExternalId",
```
Comment (galales) [inline src/test/resources/authz.json:35]: this requires also the test in the spec

**Review by galales — CHANGES_REQUESTED** [review]

File: src/test/resources/authz.json (hunk 2 above)
Comment (nttdata-rtorsoli) [reply src/test/resources/authz.json:35]: I prefer to add this test after merged with PIN-3494 because I have created an ad-hoc spec for this purpose.

Comment (galales) [reply src/test/resources/authz.json:35]: can you move the PR to draft then? 🙏

Comment (nttdata-rtorsoli) [reply src/test/resources/authz.json:35]: Resolved!

**Review by TonioGela — APPROVED** [review]

**Review by galales — APPROVED** [review]

**Review by beetlecrunch — APPROVED** [review]

**Pull request merged**